disk_usage/disk_usage.py

```diff
@@ -69,23 +63,28 @@ def main():
             count += 1
             if count % 10**6 == 0:
                 logging.info(f'{count // 10**6} M blobs...')
-            name = f'gs://{bucket_name}/{aggregate_level(blob.name)}'
-            stats = aggregate_stats[name]
-            stats.size += blob.size
-            stats.num_blobs += 1
+            folder = aggregate_level(blob.name)
+            last_index = 0
+            while True:
+                index = folder.find('/', last_index)
+                substr = folder[:index] if index != -1 else folder
+                path = f'gs://{bucket_name}/{substr}'
+                stats = aggregate_stats[path]
+                stats['size'] += blob.size
+                stats['num_blobs'] += 1
```

Comment:
Stylistically I prefer setting on the instance, but interested for this many millions if it makes a substantial difference.

```suggestion
                aggregate_stats[path]['size'] += blob.size
                aggregate_stats[path]['num_blobs'] += 1
```

Reply:
Probably doesn't make a huge difference overall.